components/select/index.tsx

```diff
@@ -190,24 +196,30 @@ export default class Select extends React.Component<SelectProps, {}> {
         <Icon type="down" className={`${prefixCls}-arrow-icon`} />
       );
 
-    const removeIcon = (
+    const finalRemoveIcon = removeIcon && (
+      React.isValidElement<{ className?: string }>(removeIcon)
+        ? React.cloneElement(removeIcon) : removeIcon) || (
       <Icon type="close" className={`${prefixCls}-remove-icon`} />
     );
 
-    const clearIcon = (
+    const finalClearIcon = clearIcon && (
+      React.isValidElement<{ className?: string }>(clearIcon)
+        ? React.cloneElement(clearIcon) : clearIcon) || (
       <Icon type="close-circle" theme="filled" className={`${prefixCls}-clear-icon`} />
     );
 
-    const menuItemSelectedIcon = (
+    const finalMenuItemSelectedIcon = menuItemSelectedIcon && (
+      React.isValidElement<{ className?: string }>(menuItemSelectedIcon)
+        ? React.cloneElement(menuItemSelectedIcon) : menuItemSelectedIcon) || (
```

Comment:
Also need pass ```className={`${prefixCls}-selected-icon`}```:
<img width="136" alt="2018-11-06 3 09 48" src="https://user-images.githubusercontent.com/5378891/48048297-05b81080-e1d6-11e8-89bc-306bc72fe0cb.png">

Reply:
After reading the code, I found the specified className of `suffixIcon` is not passed when using custom `suffixIcon`, while clear/remove/suffix icons have not their style. So, they work fine right now. In the case of the potentional style issue, all of the classNames are passed to the custom icons right now.